packages/backend-data/src/factory.ts

```diff
@@ -194,23 +198,33 @@ class DataGenerator implements ConstructContainerEntryGenerator {
       ...propsFunctions,
       ...lambdaFunctions,
     });
-    const amplifyApi = new AmplifyData(scope, this.defaultName, {
-      apiName: this.props.name,
-      definition: amplifyGraphqlDefinition,
-      authorizationModes,
-      outputStorageStrategy: this.outputStorageStrategy,
-      functionNameMap,
-      translationBehavior: {
-        sandboxModeEnabled,
-        /**
-         * The destructive updates should be always allowed in backend definition and not to be controlled on the IaC
-         * The CI/CD check should take the responsibility to validate if any tables are being replaced and determine whether to execute the changeset
-         */
-        allowDestructiveGraphqlSchemaUpdates: true,
-      },
-    });
+    let amplifyApi = undefined;
 
-    /**
+    try {
+      amplifyApi = new AmplifyData(scope, this.defaultName, {
+        apiName: this.props.name,
+        definition: amplifyGraphqlDefinition,
+        authorizationModes,
+        outputStorageStrategy: this.outputStorageStrategy,
+        functionNameMap,
+        translationBehavior: {
+          sandboxModeEnabled,
+          /**
+           * The destructive updates should be always allowed in backend definition and not to be controlled on the IaC
+           * The CI/CD check should take the responsibility to validate if any tables are being replaced and determine whether to execute the changeset
+           */
+          allowDestructiveGraphqlSchemaUpdates: true,
+        },
+      });
+    } catch (error) {
+      throw new AmplifyFault(
+        'AmplifyDataConstructFault',
+        { message: 'Failed to instantiate data construct' },
+        error as Error
+      );
+    }
```

Comment:
Couple of questions.
1. Is this necessary for this PR ? 
2. If we do this should we do this for all verticals and perhaps place this code in platform instead? 
3. What if data construct initialization is due to "user error" due to invalid inputs ? Would we be masking user errors that way? (I guess the answer is that these are today classified as faults, but I'd like to prompt discussion about how can we differentiate between faults and errors coming from data construct).

Reply:
Re. 2. - this was the case in gen1 too :-(

Reply:
It's fine then to focus this on data then...